### PR TITLE
Firebase authentication

### PR DESCRIPTION
**Pull request by msdeshmukh009**
### Firebase Authentication
- Signup
- Login 
- Logout

### Components added 
- Loading
- Overlay container

### Pages added
- Profile

### CODE REVIEW

File: src/components/navbar/Navbar.tsx [hunk 1]
```diff
@@ -1,9 +1,14 @@
 import "./navbar.css";
 import { Link } from "react-router-dom";
 import { useTheme } from "../../hooks";
+import { useAuth } from "../../hooks";
```
Comment (Chakravarthi0) [inline src/components/navbar/Navbar.tsx]: You can import both the hooks in one line

File: src/hooks/useAuth.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+            type: "SET_AUTHENTICATION",
+            payload: {
+              isAuthenticated: false,
+              uid: "",
+            },
+          });
+        }
+      })();
+    });
+  }, [authDispatch]);
```
Comment (Chakravarthi0) [inline src/hooks/useAuth.ts]: why are you using authDispatch function as dependency?

**Review by Chakravarthi0 — APPROVED** [review]
the app is working as expected.
code looks clean.
apart from the suggestions, everything looks good.

File: src/hooks/useAuth.ts [hunk 3]
```diff
@@ -0,0 +1,108 @@
+import { useContext, useEffect } from "react";
+import { useNavigate } from "react-router-dom";
+import { AuthContext } from "../context";
+import { LogInUserInput, UserDetailType } from "../types";
+import { signupService, logoutService, loginService } from "../services";
+import { auth } from "../firebase-config";
+import { createUserDocument } from "../utils";
+import { getErrorMessage } from "../utils";
```
Comment (gopalbbharadva) [inline src/hooks/useAuth.ts]: It can be one liner import.

**Review by gopalbbharadva — APPROVED** [review]
Code looks fine, Feature working fine. nice work.

**Review by manojasarsa — APPROVED** [review]
Authentication is working fine.
GTG!

File: src/hooks/useAuth.ts (hunk 2 above)
Comment (msdeshmukh009) [reply src/hooks/useAuth.ts]: It was giving me waring that's why. I am looking into how to fix the issue. Thanks for the review.

File: src/hooks/useAuth.ts (hunk 3 above)
Comment (msdeshmukh009) [reply src/hooks/useAuth.ts]: Yep. Thanks. Fixed it.

**Pull request merged**